sdk/keccak-hasher/src/lib.rs

```diff
@@ -1,34 +1,45 @@
 //! Hashing with the [keccak] (SHA-3) hash function.
 //!
 //! [keccak]: https://keccak.team/keccak.html
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![cfg_attr(feature = "frozen-abi", feature(min_specialization))]
 
 #[cfg(feature = "borsh")]
 use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+#[cfg(any(feature = "sha3", not(target_os = "solana")))]
+use sha3::{Digest, Keccak256};
+pub use solana_hash::{ParseHashError, HASH_BYTES, MAX_BASE58_LEN};
 use {
-    sha3::{Digest, Keccak256},
     solana_sanitize::Sanitize,
-    std::{convert::TryFrom, fmt, str::FromStr},
-    thiserror::Error,
+    std::{fmt, str::FromStr},
 };
 
-pub const HASH_BYTES: usize = 32;
-/// Maximum string length of a base58 encoded hash
-const MAX_BASE58_LEN: usize = 44;
-#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
+// TODO: replace this with `solana_hash::Hash` in the
+// next breaking change.
+// It's a breaking change because the field is public
+// here and private in `solana_hash`, and making
+// it public in `solana_hash` would break wasm-bindgen
```

Comment:
Someone tried to combine all the hashes awhile ago, and we couldn't for this exact reason :upside_down_face: